mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -907,6 +910,27 @@ public enum ShutdownOption : int {
 		SHUT_RDWR = 0x03,   /* No more receptions or transmissions. */
 	}
 
+	// Used by libMonoPosixHelper to distinguish between different sockaddr types
+	[Map]
+	enum SockaddrType : int {
+		Invalid,
+		SockaddrStorage,
+		SockaddrUn,
+		Sockaddr,
+		SockaddrIn,
+		SockaddrIn6,
+
+		// Flag to indicate that this Sockaddr must be wrapped with a _SockaddrDynamic wrapper
+		MustBeWrapped = 0x8000,
```

Comment:
The `enum SockaddrType` is internal, so none of the `SockaddrType` values should be exposed.